Pocket/pocket-ios
# feat(tags): add recent tags

## Summary
Add Recent Tags section to both Edit and Add Tags screen

* This work does not include guarding recent tags with premium check (this will be done in another PR)

## References 
IN-1268

## Implementation Details
Created `RecentTag` component and add a case of `recent(String)`  to enum `TagType` . Added `x` to the input tags in Add Tags view. Created a `RecentTagProvider` class to hold the logic of recent tags to be used across the three view models related to tags `PocketAddTagsViewModel`, `SaveToAddTagsViewModel` and `TagsFilterViewModel`.

Moved some files that were related to tags to `SharedPocketKit` instead of `Textile` as it made more sense for it to belong there.


## Test Steps
Verify the AC below: 

There should be two sections: (1) Recent Tags, which shows the three most-recently used tags, and (2) All Tags, which shows all of a user’s tags. Both sections should be sorted alphabetically. 
1. Recent Tags for both screens should be consistent 
2. Recent Tags should appear first in the tags list 
3. Recent Tags should be distinguishable in some way 
4. Recent Tags should be sorted by most-recent use (adding a tag to an item)
5. All Tags should be sorted by alphabetical order
6. Add x to right side of tag for Add Tags screen

## PR Checklist:

## Screenshots

https://user-images.githubusercontent.com/6743397/230919581-294613cc-875c-4130-a432-508d64beb26b.mp4
